src/fundus/scraping/pipeline.py

```diff
@@ -161,81 +193,25 @@ def __init__(self, *publishers: Union[PublisherEnum, Type[PublisherEnum]]):
-                articles are returned. Defaults to None.
             restrict_sources_to (Optional[List[Literal["rss", "sitemap", "news"]]]): Let's you restrict
                 sources defined in the publisher specs. If set, only articles from given source types
                 will be yielded.
-            error_handling (Literal["suppress", "catch", "raise"]): Define how to handle errors
-                encountered during extraction. If set to "suppress", all errors will be skipped, either
-                with None values for respective attributes in the extraction or by skipping entire articles.
-                If set to "catch", errors will be caught as attribute values or, if an entire article fails,
-                through Article.exception. If set to "raise" all errors encountered during extraction will
-                be raised. Defaults to "suppress".
-            only_complete (Union[bool, ExtractionFilter]): Set extraction filters. If False, all articles
-                will be yielded, if True, only complete ones. Defaults to True. See the docs for more
-                information about ExtractionFilter.
-            delay (Optional[Union[float, Delay]]): Set a delay time in seconds to be used between article
-                batches. You can set a delay directly using float or any callable satisfying the Delay
-                protocol. If set to None, no delay will be used between batches. See Delay for more
-                information. Defaults to None.
-            url_filter (Optional[URLFilter]): A callable object satisfying the URLFilter protocol to skip
-                URLs before download. This filter applies on both requested and responded URL. Defaults to None.
-            only_unique (bool): If set to True, articles yielded will be unique on the responded URL.
-                Always returns the first encountered article. the Defaults to True.
-
-        Returns:
-            Iterator[Article]: An iterator yielding objects of type Article.
         """
 
-        extraction_filter: Optional[ExtractionFilter]
-
-        if isinstance(only_complete, bool):
-            extraction_filter = (
-                None
-                if only_complete is False
-                else lambda extracted: not all(
-                    bool(v) if not isinstance(v, Exception) else False for k, v in extracted.items()
-                )
-            )
-        else:
-            extraction_filter = only_complete
-
-        if isinstance(delay, float):
-
-            def constant_delay() -> float:
-                return delay  # type: ignore[return-value]
-
-            delay = constant_delay
+        if not publishers:
+            raise ValueError("param <publishers> of <Crawler.__init__> has to be non empty")
+        nested_publisher = [listify(publisher) for publisher in publishers]
+        flat_publishers: Set[PublisherEnum] = set(more_itertools.flatten(nested_publisher))
```

Comment:
I think we could also use collapse from more itertools to skip the listify part.

```python
>>> from more_itertools import collapse
>>> list(collapse([[1, 2], 3]))
[1, 2, 3]
```

Reply:
Ah nice, thanks :)